# Perovskite Symmetry-Breaking Example
*October 28, 2025*

This notebook demonstrates the importance of initially breaking the structural symmetry as well as considering larger supercells for $SrBO_3$ perovskites in DFT calculations. We will do this using the $SrTiO_3$, $SrVO_3$, and $SrNbO_3$ calculations as an example, however it is important to consider in a variety of systems as demonstrated by Alex Zunger's group as well as our previous exploration of transparent conducting high-entropy perovskite oxides [(Link)](https://advanced.onlinelibrary.wiley.com/doi/full/10.1002/advs.202509868).

Oftentimes DFT calculations are only performed for perovskite oxides with a 5-atom cubic unit cell. These smallest representations do not allow for symmetry-distortions such as the well-known octahedral tilting common in perovskites to occur during the relaxation process. The discrepancies between DFT predictions and experiment is commonly estimated as the "correlation" in these systems, however by allowing for these symmetry-breaking distortions (especially when accounted for both in atomic structure and magnetic ordering). We will only consider the structural aspect here with both FM and AFM-G magnetic ordering as an example, with AFM-G being the highest possible "correlation" due to each B-cation only seeing B-cations with opposite spin.

# $SrTiO_3$
This composition is known as one of the sterotypical peroskite oxides, so we will start here. We have confidence that know $Ti^{4+}$ has no unpaired valence electrons and can therefore run our calculations without spin polarization.

In [1]:
from pymatgen.core import Structure
from pytheos.structure import generation, utils

# 5-atom perovskite unit cell
SrTiO3_unitcell = Structure.from_file("perovskite_files/SrTiO3_unitcell.vasp")
print(SrTiO3_unitcell)

# 40-atom perovskite supercell
SrTiO3_2x2x2 = generation.make_supercell(struc=SrTiO3_unitcell, dimensions=[2, 2, 2])

Full Formula (Sr1 Ti1 O3)
Reduced Formula: SrTiO3
abc   :   4.000000   4.000000   4.000000
angles:  90.000000  90.000000  90.000000
pbc   :       True       True       True
Sites (5)
  #  SP      a    b    c
---  ----  ---  ---  ---
  0  Sr    0    0    0
  1  Ti    0.5  0.5  0.5
  2  O     0.5  0    0.5
  3  O     0.5  0.5  0
  4  O     0    0.5  0.5


In [2]:
from pytheos.vasp import inputs
import os

In [7]:
os.mkdir("perovskite_files/SrTiO3_unitcell")

SrTiO3_unitcell_inputs = inputs.CalcInputs(struc=SrTiO3_unitcell)
SrTiO3_unitcell_inputs.update_incar({"ISPIN": 1, "ISIF": 8}) # turn off spin polarization, and maintain cubic cell symmetry and relax atom positions and cell volume
SrTiO3_unitcell_inputs.write_files("perovskite_files/SrTiO3_unitcell/relaxation")

os.system("cp ../scripts/submitvasp perovskite_files/SrTiO3_unitcell/relaxation")
os.system("cp ../scripts/cstdn_vasp_doublerelax.py perovskite_files/SrTiO3_unitcell/relaxation/cstdn.py")

# we are going to rattle the initial atomic positions for consistency with the supercell calculation
os.system("cp perovskite_files/SrTiO3_unitcell/relaxation/POSCAR perovskite_files/SrTiO3_unitcell/relaxation/POSCAR_unrattled")
poscar = Structure.from_file(filename="perovskite_files/SrTiO3_unitcell/relaxation/POSCAR")
poscar_rattled = utils.rattle_atoms(struc=poscar)
poscar_rattled.to_file("perovskite_files/SrTiO3_unitcell/relaxation/POSCAR")
print(poscar_rattled)

Full Formula (Sr1 Ti1 O3)
Reduced Formula: SrTiO3
abc   :   4.000000   4.000000   4.000000
angles:  90.000000  90.000000  90.000000
pbc   :       True       True       True
Sites (5)
  #  SP            a          b         c
---  ----  ---------  ---------  --------
  0  Sr    -6.7e-05   -0.001723  0.00224
  1  Ti     0.498339   0.502553  0.497735
  2  O      0.501813  -0.002333  0.499983
  3  O      0.498348   0.503019  0.001972
  4  O     -0.002595   0.502137  0.504278


In [8]:
os.mkdir("perovskite_files/SrTiO3_2x2x2")

SrTiO3_2x2x2_inputs = inputs.CalcInputs(struc=SrTiO3_2x2x2)
SrTiO3_2x2x2_inputs.update_incar({"ISPIN": 1, "ISIF": 8}) # turn off spin polarization, and maintain cubic cell symmetry and relax atom positions and cell volume
SrTiO3_2x2x2_inputs.write_files("perovskite_files/SrTiO3_2x2x2/relaxation")

os.system("cp ../scripts/submitvasp perovskite_files/SrTiO3_2x2x2/relaxation")
os.system("cp ../scripts/cstdn_vasp_doublerelax.py perovskite_files/SrTiO3_2x2x2/relaxation/cstdn.py")

# we are going to rattle the initial atomic positions for consistency with the supercell calculation
os.system("cp perovskite_files/SrTiO3_2x2x2/relaxation/POSCAR perovskite_files/SrTiO3_2x2x2/relaxation/POSCAR_unrattled")
poscar = Structure.from_file("perovskite_files/SrTiO3_2x2x2/relaxation/POSCAR")
poscar_rattled = utils.rattle_atoms(struc=poscar)
poscar_rattled.to_file("perovskite_files/SrTiO3_2x2x2/relaxation/POSCAR")
print(poscar_rattled)

Full Formula (Sr8 Ti8 O24)
Reduced Formula: SrTiO3
abc   :   8.000000   8.000000   8.000000
angles:  90.000000  90.000000  90.000000
pbc   :       True       True       True
Sites (40)
  #  SP            a          b          c
---  ----  ---------  ---------  ---------
  0  Sr    -0.00059   -0.000304   0.001831
  1  Sr     0.000257   0.00036    0.499558
  2  Sr    -0.001343   0.499885   0.001019
  3  Sr     0.001981   0.499264   0.499013
  4  Sr     0.497763   0.000826   0.000256
  5  Sr     0.501837  -0.000669   0.499578
  6  Sr     0.5006     0.49847    0.001428
  7  Sr     0.500607   0.49988    0.499406
  8  Ti     0.249813   0.248485   0.249145
  9  Ti     0.24827    0.250823   0.747332
 10  Ti     0.249278   0.751232   0.250432
 11  Ti     0.250623   0.748889   0.750867
 12  Ti     0.749828   0.249728   0.249779
 13  Ti     0.749956   0.249597   0.751341
 14  Ti     0.749416   0.746492   0.249588
 15  Ti     0.749747   0.74997    0.75221
 16  O      0.250412  -0.000748   0.249135

/storage/work/jts6114/.conda/envs/pytheos/lib/python3.12/site-packages/pymatgen/io/vasp/sets.py:486: BadInputSetWarning: POTCAR data with symbol Sr_sv is not known by pymatgen to correspond with the selected user_potcar_functional='PBE_54'. This POTCAR is known to correspond with functionals ['PBE_64', 'PBE_52_W_HASH', 'PBE_54_W_HASH']. Please verify that you are using the right POTCARs!
  potcar="\n".join(self.potcar_symbols) if potcar_spec else self.potcar,
/storage/work/jts6114/.conda/envs/pytheos/lib/python3.12/site-packages/pymatgen/io/vasp/sets.py:486: BadInputSetWarning: POTCAR data with symbol Ti_pv is not known by pymatgen to correspond with the selected user_potcar_functional='PBE_54'. This POTCAR is known to correspond with functionals ['PBE_64', 'PBE_52_W_HASH', 'PBE_54_W_HASH']. Please verify that you are using the right POTCARs!
  potcar="\n".join(self.potcar_symbols) if potcar_spec else self.potcar,
/storage/work/jts6114/.conda/envs/pytheos/lib/python3.12/site-packages/p

In [9]:
# again we should run DOS calculations to be safe in our comparison...
from pytheos.vasp.modifiers import CalcModifier

SrTiO3_unitcell_calc = CalcModifier(source_dir="perovskite_files/SrTiO3_unitcell/relaxation")
SrTiO3_unitcell_calc.to_dos()
SrTiO3_unitcell_calc.write_files(output_dir="perovskite_files/SrTiO3_unitcell/dos")
os.system("cp ../scripts/cstdn_vasp.py perovskite_files/SrTiO3_unitcell/dos/cstdn.py")

SrTiO3_2x2x2_calc = CalcModifier(source_dir="perovskite_files/SrTiO3_2x2x2/relaxation")
SrTiO3_2x2x2_calc.to_dos()
SrTiO3_2x2x2_calc.write_files(output_dir="perovskite_files/SrTiO3_2x2x2/dos")
os.system("cp ../scripts/cstdn_vasp.py perovskite_files/SrTiO3_2x2x2/dos/cstdn.py")

0

When we take a look at the energetic and electronic structure differences between both the 5-atom unit cell and the the 40-atom supercell for $SrTiO_3$, we can see that these calculations are basically equivalent: they both return to nearly the same energy regardless of using the 5-atom unit cell or the larger 40-atom supercell.
- there are some differences that can be observed in the density of states plots below, however they are qualitatively the same and are likely just the result of a less-tight structural relaxation convergence and can be considered the same for our purposes in this example...

In [12]:
from pytheos.vasp import outputs

SrTiO3_unitcell = outputs.CalcOutputs(source_dir="perovskite_files/SrTiO3_unitcell/dos")
SrTiO3_2x2x2 = outputs.CalcOutputs(source_dir="perovskite_files/SrTiO3_2x2x2/dos")

print("ENERGY (eV/atom)")
print(f"\tunitcell: {SrTiO3_unitcell.final_energy_per_atom:.4f}")
print(f"\tsupercell: {SrTiO3_2x2x2.final_energy_per_atom:.4f}")

print("ELECTRONIC BAND GAP (eV)")
print(f"\tunitcell: {SrTiO3_unitcell.band_gap:.2f}")
print(f"\tsupercell: {SrTiO3_2x2x2.band_gap:.2f}")

ENERGY (eV/atom)
	unitcell: -12.6474
	supercell: -12.6476
ELECTRONIC BAND GAP (eV)
	unitcell: 2.30
	supercell: 2.24


**unitcell**
<div>
<img src="perovskite_files/SrTiO3_unitcell/dos/dos.png" width="500"/>
</div>

**supercell**
<div>
<img src="perovskite_files/SrTiO3_2x2x2/dos/dos.png" width="500"/>
</div>

We can also take a look at the local atomic structure for the relaxed structures in each using the `extract_octahedral_bondangles` function in `pytheos`
- since this function does not take into account periodic boundary conditions, we will make each relaxed structure into a 2x2x2 supercell of itself

In [14]:
from pytheos.structure import analysis

SrTiO3_unitcell_doubled = generation.make_supercell(SrTiO3_unitcell.structure, [2, 2, 2])
SrTiO3_2x2x2_doubled = generation.make_supercell(SrTiO3_2x2x2.structure, [2, 2, 2])

In [15]:
analysis.extract_octahedral_bondangles(struc=SrTiO3_unitcell_doubled, bsite_cations=["Ti"])

Extracting average octahedral bond angles...
bsite cations: ['Ti']
Ti
	Ti1(#8) - O1(#17) - Ti1(#9) -> 179.2°
	Ti1(#8) - O1(#22) - Ti1(#10) -> 179.3°
	Ti1(#8) - O1(#30) - Ti1(#12) -> 179.3°
	Ti1(#9) - O1(#25) - Ti1(#11) -> 179.3°
	Ti1(#9) - O1(#33) - Ti1(#13) -> 179.3°
	Ti1(#10) - O1(#23) - Ti1(#11) -> 179.2°
	Ti1(#10) - O1(#36) - Ti1(#14) -> 179.3°
	Ti1(#11) - O1(#39) - Ti1(#15) -> 179.3°
	Ti1(#12) - O1(#29) - Ti1(#13) -> 179.2°
	Ti1(#12) - O1(#34) - Ti1(#14) -> 179.3°
	Ti1(#13) - O1(#37) - Ti1(#15) -> 179.3°
	Ti1(#14) - O1(#35) - Ti1(#15) -> 179.2°
num bond angles found: 12
avg bond angle: 179.3°
elapsed time: 0.06 seconds


atom1 atom1_species  atom2 atom2_species  atom3 atom3_species  bondangle
0       8            Ti     17            O2      9            Ti    179.218
1       8            Ti     22            O2     10            Ti    179.263
2       8            Ti     30            O2     12            Ti    179.345
3       9            Ti     25            O2     11            Ti    179.263
4       9            Ti     33            O2     13            Ti    179.345
5      10            Ti     23            O2     11            Ti    179.218
6      10            Ti     36            O2     14            Ti    179.345
7      11            Ti     39            O2     15            Ti    179.345
8      12            Ti     29            O2     13            Ti    179.218
9      12            Ti     34            O2     14            Ti    179.263
10     13            Ti     37            O2     15            Ti    179.263
11     14            Ti     35            O2     15            Ti    179.218

In [16]:
analysis.extract_octahedral_bondangles(struc=SrTiO3_2x2x2_doubled, bsite_cations=["Ti"])

Extracting average octahedral bond angles...
bsite cations: ['Ti']
Ti
	Ti1(#64) - O1(#132) - Ti1(#65) -> 179.8°
	Ti1(#64) - O1(#134) - Ti1(#66) -> 177.7°
	Ti1(#64) - O1(#142) - Ti1(#68) -> 178.5°
	Ti1(#65) - O1(#137) - Ti1(#67) -> 179.5°
	Ti1(#65) - O1(#145) - Ti1(#69) -> 177.9°
	Ti1(#65) - O1(#153) - Ti1(#72) -> 177.8°
	Ti1(#66) - O1(#138) - Ti1(#67) -> 177.9°
	Ti1(#66) - O1(#148) - Ti1(#70) -> 178.1°
	Ti1(#66) - O1(#176) - Ti1(#80) -> 179.6°
	Ti1(#67) - O1(#151) - Ti1(#71) -> 178.6°
	Ti1(#67) - O1(#159) - Ti1(#74) -> 179.8°
	Ti1(#67) - O1(#179) - Ti1(#81) -> 177.7°
	Ti1(#68) - O1(#144) - Ti1(#69) -> 177.8°
	Ti1(#68) - O1(#146) - Ti1(#70) -> 179.6°
	Ti1(#68) - O1(#226) - Ti1(#96) -> 178.0°
	Ti1(#69) - O1(#149) - Ti1(#71) -> 177.6°
	Ti1(#69) - O1(#165) - Ti1(#76) -> 179.9°
	Ti1(#69) - O1(#229) - Ti1(#97) -> 178.6°
	Ti1(#70) - O1(#150) - Ti1(#71) -> 179.8°
	Ti1(#70) - O1(#188) - Ti1(#84) -> 177.7°
	Ti1(#70) - O1(#232) - Ti1(#98) -> 178.6°
	Ti1(#71) - O1(#171) - Ti1(#78) -> 177.8°
	Ti1(#

atom1 atom1_species  atom2 atom2_species  atom3 atom3_species  bondangle
0       64            Ti    132            O2     65            Ti    179.794
1       64            Ti    134            O2     66            Ti    177.739
2       64            Ti    142            O2     68            Ti    178.544
3       65            Ti    137            O2     67            Ti    179.477
4       65            Ti    145            O2     69            Ti    177.924
..     ...           ...    ...           ...    ...           ...        ...
139    123            Ti    319            O2    127            Ti    178.592
140    124            Ti    312            O2    125            Ti    177.787
141    124            Ti    314            O2    126            Ti    179.564
142    125            Ti    317            O2    127            Ti    177.630
143    126            Ti    318            O2    127            Ti    179.846

[144 rows x 7 columns]

Notice that the 40-atom supercell has slightly tilted B-site octahedra compared to the 5-atom unit cell as observed through the B-site octahedral bond angles, however as these energies are effectively degenerate these can be considered as equally stable. Compared to the following example, these values are quite close and can be considered equivalent...

# $SrNbO3$
Now let's take a look at a different B-site cation: $Nb$. We will use the same exact process as what we did for $SrTiO_3$, however swapping out the B-site cation from $Ti$ for $Nb$. We will however include spin polarization (just FM initial ordering for this example here).

In [18]:
# 5-atom perovskite unit cell
SrNbO3_unitcell = Structure.from_file("perovskite_files/SrNbO3_unitcell.vasp")
print(SrNbO3_unitcell)

# 40-atom perovskite supercell
SrNbO3_2x2x2 = generation.make_supercell(struc=SrNbO3_unitcell, dimensions=[2, 2, 2])


# calculation setup for SrNbO3 unit cell
os.mkdir("perovskite_files/SrNbO3_unitcell")

SrNbO3_unitcell = inputs.CalcInputs(struc=SrNbO3_unitcell)
SrNbO3_unitcell.update_incar({"ISIF": 8}) # maintain cubic cell symmetry and relax atom positions and cell volume
SrNbO3_unitcell.write_files("perovskite_files/SrNbO3_unitcell/relaxation")

os.system("cp ../scripts/submitvasp perovskite_files/SrNbO3_unitcell/relaxation")
os.system("cp ../scripts/cstdn_vasp_doublerelax.py perovskite_files/SrNbO3_unitcell/relaxation/cstdn.py")

# we are going to rattle the initial atomic positions for consistency with the supercell calculation
os.system("cp perovskite_files/SrNbO3_unitcell/relaxation/POSCAR perovskite_files/SrNbO3_unitcell/relaxation/POSCAR_unrattled")
poscar = Structure.from_file("perovskite_files/SrNbO3_unitcell/relaxation/POSCAR")
poscar_rattled = utils.rattle_atoms(struc=poscar)
poscar_rattled.to_file("perovskite_files/SrNbO3_unitcell/relaxation/POSCAR")
print(poscar_rattled)


# calculation setup for SrNbO3 supercell
os.mkdir("perovskite_files/SrNbO3_2x2x2")

SrNbO3_2x2x2_inputs = inputs.CalcInputs(struc=SrNbO3_2x2x2)
SrNbO3_2x2x2_inputs.update_incar({"ISIF": 8}) # maintain cubic cell symmetry and relax atom positions and cell volume
SrNbO3_2x2x2_inputs.write_files("perovskite_files/SrNbO3_2x2x2/relaxation")

os.system("cp ../scripts/submitvasp perovskite_files/SrNbO3_2x2x2/relaxation")
os.system("cp ../scripts/cstdn_vasp_doublerelax.py perovskite_files/SrNbO3_2x2x2/relaxation/cstdn.py")

# we are going to rattle the initial atomic positions for consistency with the supercell calculation
os.system("cp perovskite_files/SrNbO3_2x2x2/relaxation/POSCAR perovskite_files/SrNbO3_2x2x2/relaxation/POSCAR_unrattled")
poscar = Structure.from_file("perovskite_files/SrNbO3_2x2x2/relaxation/POSCAR")
poscar_rattled = utils.rattle_atoms(struc=poscar)
poscar_rattled.to_file("perovskite_files/SrNbO3_2x2x2/relaxation/POSCAR")
print(poscar_rattled)

Full Formula (Sr1 Nb1 O3)
Reduced Formula: SrNbO3
abc   :   4.000000   4.000000   4.000000
angles:  90.000000  90.000000  90.000000
pbc   :       True       True       True
Sites (5)
  #  SP      a    b    c
---  ----  ---  ---  ---
  0  Sr    0    0    0
  1  Nb    0.5  0.5  0.5
  2  O     0.5  0    0.5
  3  O     0.5  0.5  0
  4  O     0    0.5  0.5
Full Formula (Sr1 Nb1 O3)
Reduced Formula: SrNbO3
abc   :   4.000000   4.000000   4.000000
angles:  90.000000  90.000000  90.000000
pbc   :       True       True       True
Sites (5)
  #  SP            a          b          c
---  ----  ---------  ---------  ---------
  0  Sr     0.000954   0.00333   -0.002199
  1  Nb     0.49781    0.502213   0.503411
  2  O      0.499257  -0.004032   0.496889
  3  O      0.49946    0.498509   0.003347
  4  O     -0.004348   0.499153   0.500186
Full Formula (Sr8 Nb8 O24)
Reduced Formula: SrNbO3
abc   :   8.000000   8.000000   8.000000
angles:  90.000000  90.000000  90.000000
pbc   :       True       True

In [19]:
# again we should run DOS calculations to be safe in our comparison...
import os
from pytheos.vasp.modifiers import CalcModifier

SrNbO3_unitcell_calc = CalcModifier(source_dir="perovskite_files/SrNbO3_unitcell/relaxation")
SrNbO3_unitcell_calc.to_dos()
SrNbO3_unitcell_calc.write_files(output_dir="perovskite_files/SrNbO3_unitcell/dos")
os.system("cp ../scripts/cstdn_vasp.py perovskite_files/SrNbO3_unitcell/dos/cstdn.py")

SrNbO3_2x2x2_calc = CalcModifier(source_dir="perovskite_files/SrNbO3_2x2x2/relaxation")
SrNbO3_2x2x2_calc.to_dos()
SrNbO3_2x2x2_calc.write_files(output_dir="perovskite_files/SrNbO3_2x2x2/dos")
os.system("cp ../scripts/cstdn_vasp.py perovskite_files/SrNbO3_2x2x2/dos/cstdn.py")

0

Lets again first start by looking at the energy (relative stability) and electronic band gap for both SrNbO3 unit cell and 2x2x2 supercell calculations...

In [1]:
from pytheos.vasp import outputs

SrNbO3_unitcell = outputs.CalcOutputs(source_dir="perovskite_files/SrNbO3_unitcell/dos")
SrNbO3_2x2x2 = outputs.CalcOutputs(source_dir="perovskite_files/SrNbO3_2x2x2/dos")

print("ENERGY (eV/atom)")
print(f"\tunitcell: {SrNbO3_unitcell.final_energy_per_atom:.4f}")
print(f"\tsupercell: {SrNbO3_2x2x2.final_energy_per_atom:.4f}")

print("ELECTRONIC BAND GAP (eV)")
print(f"\tunitcell: {SrNbO3_unitcell.band_gap:.2f}")
print(f"\tsupercell: {SrNbO3_2x2x2.band_gap:.2f}")

ENERGY (eV/atom)
	unitcell: -14.5424
	supercell: -14.5669
ELECTRONIC BAND GAP (eV)
	unitcell: 0.00
	supercell: 0.00


Notice that while we obtain metallic band gaps for both calculations, we obtain a lower energy for the supercell calculation compared to the unit cell calculation by 24.5 meV/atom...

Let's compared the DOS for each calculation as well.

**unitcell**
<div>
<img src="perovskite_files/SrNbO3_unitcell/dos/dos.png" width="500"/>
</div>

**supercell**
<div>
<img src="perovskite_files/SrNbO3_2x2x2/dos/dos.png" width="500"/>
</div>

Since the electronic structures are quite similar, let's take a look into the local atomic structure to explore this energy differences. Specifically, let's compared the octahedral bond angles like we did for the SrTiO3 unit cell and supercell calculations.
- again, we are going to make 2x2x2 supercells for each relaxed structure as the function we are going to use does *not* account for periodic boundary conditions

In [3]:
from pytheos.structure import analysis, generation

SrNbO3_unitcell_doubled = generation.make_supercell(SrNbO3_unitcell.structure, [2, 2, 2])
SrNbO3_2x2x2_doubled = generation.make_supercell(SrNbO3_2x2x2.structure, [2, 2, 2])

In [4]:
analysis.extract_octahedral_bondangles(struc=SrNbO3_unitcell_doubled, bsite_cations=["Nb"])

Extracting average octahedral bond angles...
bsite cations: ['Nb']
Nb
	Nb1(#8) - O1(#18) - Nb1(#12) -> 179.9°
	Nb1(#8) - O1(#20) - Nb1(#9) -> 180.0°
	Nb1(#8) - O1(#22) - Nb1(#10) -> 179.9°
	Nb1(#9) - O1(#21) - Nb1(#13) -> 179.9°
	Nb1(#9) - O1(#25) - Nb1(#11) -> 179.9°
	Nb1(#10) - O1(#24) - Nb1(#14) -> 179.9°
	Nb1(#10) - O1(#26) - Nb1(#11) -> 180.0°
	Nb1(#11) - O1(#27) - Nb1(#15) -> 179.9°
	Nb1(#12) - O1(#32) - Nb1(#13) -> 180.0°
	Nb1(#12) - O1(#34) - Nb1(#14) -> 179.9°
	Nb1(#13) - O1(#37) - Nb1(#15) -> 179.9°
	Nb1(#14) - O1(#38) - Nb1(#15) -> 180.0°
num bond angles found: 12
avg bond angle: 179.9°
elapsed time: 0.06 seconds


atom1 atom1_species  atom2 atom2_species  atom3 atom3_species  bondangle
0       8            Nb     18            O2     12            Nb    179.908
1       8            Nb     20            O2      9            Nb    179.957
2       8            Nb     22            O2     10            Nb    179.935
3       9            Nb     21            O2     13            Nb    179.908
4       9            Nb     25            O2     11            Nb    179.935
5      10            Nb     24            O2     14            Nb    179.908
6      10            Nb     26            O2     11            Nb    179.957
7      11            Nb     27            O2     15            Nb    179.908
8      12            Nb     32            O2     13            Nb    179.957
9      12            Nb     34            O2     14            Nb    179.935
10     13            Nb     37            O2     15            Nb    179.935
11     14            Nb     38            O2     15            Nb    179.957

In [5]:
analysis.extract_octahedral_bondangles(struc=SrNbO3_2x2x2_doubled, bsite_cations=["Nb"])

Extracting average octahedral bond angles...
bsite cations: ['Nb']
Nb
	Nb1(#64) - O1(#132) - Nb1(#65) -> 163.7°
	Nb1(#64) - O1(#134) - Nb1(#66) -> 163.7°
	Nb1(#64) - O1(#142) - Nb1(#68) -> 165.3°
	Nb1(#65) - O1(#129) - Nb1(#72) -> 163.7°
	Nb1(#65) - O1(#137) - Nb1(#67) -> 163.7°
	Nb1(#65) - O1(#145) - Nb1(#69) -> 165.3°
	Nb1(#66) - O1(#138) - Nb1(#67) -> 163.7°
	Nb1(#66) - O1(#148) - Nb1(#70) -> 165.3°
	Nb1(#66) - O1(#176) - Nb1(#80) -> 163.7°
	Nb1(#67) - O1(#151) - Nb1(#71) -> 165.2°
	Nb1(#67) - O1(#159) - Nb1(#74) -> 163.6°
	Nb1(#67) - O1(#179) - Nb1(#81) -> 163.7°
	Nb1(#68) - O1(#130) - Nb1(#96) -> 165.4°
	Nb1(#68) - O1(#144) - Nb1(#69) -> 163.8°
	Nb1(#68) - O1(#146) - Nb1(#70) -> 163.8°
	Nb1(#69) - O1(#133) - Nb1(#97) -> 165.3°
	Nb1(#69) - O1(#149) - Nb1(#71) -> 163.8°
	Nb1(#69) - O1(#165) - Nb1(#76) -> 163.8°
	Nb1(#70) - O1(#150) - Nb1(#71) -> 163.8°
	Nb1(#70) - O1(#188) - Nb1(#84) -> 163.8°
	Nb1(#70) - O1(#232) - Nb1(#98) -> 165.2°
	Nb1(#71) - O1(#139) - Nb1(#99) -> 165.2°
	Nb1(#

atom1 atom1_species  atom2 atom2_species  atom3 atom3_species  bondangle
0       64            Nb    132            O2     65            Nb    163.663
1       64            Nb    134            O2     66            Nb    163.677
2       64            Nb    142            O2     68            Nb    165.328
3       65            Nb    129            O2     72            Nb    163.715
4       65            Nb    137            O2     67            Nb    163.680
..     ...           ...    ...           ...    ...           ...        ...
139    123            Nb    319            O2    127            Nb    165.168
140    124            Nb    312            O2    125            Nb    163.841
141    124            Nb    314            O2    126            Nb    163.807
142    125            Nb    317            O2    127            Nb    163.775
143    126            Nb    318            O2    127            Nb    163.796

[144 rows x 7 columns]

While the relaxed SrNbO3 unit cell (5-atoms) has b-site octahedral bond angles with an average of 179.9°, the SrNbO3 2x2x2 supercells has significant octahedral tilting with an average b-site bond angle of 164.2°! Let's visualize this using VESTA...

## Relaxed SrNbO3 unit cell
<div>
<img src="perovskite_files/SrNbO3_unitcell/relaxation/CONTCAR.png" width="500"/>
</div>

## Relaxed SrNbO3 2x2x2 supercell
<div>
<img src="perovskite_files/SrNbO3_2x2x2/relaxation/CONTCAR.png" width="500"/>
</div>

Importantly, the 2x2x2 supercell is large enough to allow for these symmetry-breaking phenomena (such as octahedral tilting, or anti-/para-magnetic ordering). It is also important to initially rattle the structure to allow for these lower-energy distortions to be found during the relaxation process (and to include `ISYM = 0` within the INCAR, as is default within `pytheos`). In my experience, one always rattle the initial structures to assist in finding the lower-energy, relaxed structures. These lower-symmetry structures can have a significant impact on the electronic and optical properties for perovskite systems as well (see our Advanced Science publication: [Chemically-Disordered Transparent Conductive Perovskites With High Crystalline Fidelity](https://advanced.onlinelibrary.wiley.com/doi/full/10.1002/advs.202509868), Figures S2-S4.